src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/SignatureAlgorithmFactory.php

```diff
@@ -25,19 +25,12 @@ final class SignatureAlgorithmFactory
 {
     public static function create(string $algorithm): AlgorithmInterface
     {
-        switch ($algorithm) {
-            case 'ES256':
-            case 'ES384':
-            case 'ES512':
-                if (!class_exists(Algorithm::class.'\\'.$algorithm)) {
-                    throw new \LogicException(sprintf('You cannot use the "%s" signature algorithm since "web-token/jwt-signature-algorithm-ecdsa" is not installed. Try running "composer require web-token/jwt-signature-algorithm-ecdsa".', $algorithm));
-                }
+        $algorithmFqcn = Algorithm::class.'\\'.$algorithm;
 
-                $algorithm = Algorithm::class.'\\'.$algorithm;
-
-                return new $algorithm();
+        if (!class_exists(Algorithm::class.'\\'.$algorithm)) {
```

Comment:
It is not recommended to allow all algorithms available on web-token/jwt-* as some of them are experimental, not recommended or deprecated: https://github.com/web-token/jwt-doc/blob/v2.x/the-components/encrypted-tokens-jwe/encryption-algorithms.md.

Reply:
Experimental algorithms are not in the web-token/jwt-library packages.  
But, jwk can be imposed by authentication provider. If the provider use experimental or deprecated algorithms why we couldn't use them ?

Reply:
@Spomky Invitation sent ;)